### PR TITLE
Format false branch of ternary operation

### LINKED ISSUE #3676: Indentation is too aggressive for ternary operator
Actual:

``` ts
var v =
    0 ? 1 :
        2 ? 3 :
            4;
```

Expected:

``` ts
var v =
    0 ? 1 :
    2 ? 3 :
    4;
```

I believe the fix is that if the false branch of the ternary operator should have the same indentation as the true branch. In the case where the true branch is on the same line as the condition, it should not be considered indented, and therefore the false branch should not be indented.

### PR DESCRIPTION
**Pull request by saschanaz**
~~**This PR includes #4757**~~ (#4757 is now merged :+1:)

Fixes #3676.

Current:

``` typescript
var v =
    0 ? 1 :
        2 ? 3 :
            4;
```

Fixed:

``` typescript
var v =
    0 ? 1 :
    2 ? 3 :
    4;
```

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
Sorry this didn't get looked at @SaschaNaz. Should this change have been behind a flag? My feeling is that some people might want cascading ternaries.

**Comment by basarat** [comment]
:+1: from me. Just today: 

image

As soon as a branch has a sub branch its better to move to anything other than a ternary (generally a new function containing a `switch` with cases having a `return`) so nesting is not a use case I have :rose:

**Comment by mhegazy** [comment]
sorry for the super late response. doing a quick poll, this seems a fairly subjective issue. some like their ternaries cascading, others like them on the same level. this calls for a new formatting option :)
so let's add a new option and condition the indentation on this option.

**Comment by SaschaNaz** [comment]
Okay, that's find :+1:

**Comment by SaschaNaz** [comment]
What's wrong with Linter? :/

**Comment by SaschaNaz** [comment]
Opened #13006 for linter error.

**Comment by SaschaNaz** [comment]
Just a note: the new option is now added :D

**Comment by SaschaNaz** [comment]
Rebased.

File: src/services/types.ts [hunk 1]
```diff
@@ -472,6 +472,7 @@ namespace ts {
         insertSpaceAfterOpeningAndBeforeClosingJsxExpressionBraces?: boolean;
         insertSpaceAfterTypeAssertion?: boolean;
         insertSpaceBeforeFunctionParenthesis?: boolean;
+        indentInsideTernaryOperator?: boolean;
```
Comment (mhegazy) [inline src/services/types.ts]: you also need to add this to `server\protocol.ts`

**Comment by SaschaNaz** [comment]
Renamed the option name from `indentInsideTernaryOperator` to `indentConditionalExpressionFalseBranch`, to match how existing TS codes call it.

**Pull request closed without merging**